### PR TITLE
Removed redundancy in synchronization code

### PR DESCRIPTION
**Pull request by imran-ansari**
Created a decorator that would be responsible for synchronization. Re-entrant code would need to use this decorator to acquire lock. Release would happen automatically when decorator returns.

@wdurairaj , @hpe-storage Please review the changes

### CODE REVIEW

File: hpedockerplugin/synchronization.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        def _wrapped(*a, **k):
+            call_args = inspect.getcallargs(f, *a, **k)
+            call_args['f_name'] = f.__name__
+            lck_name = lock_name.format(**call_args)
+            lock_acquired = False
+            plugin = a[0]
+            try:
+                plugin._etcd.try_lock_volname(lck_name)
+                lock_acquired = True
+                LOG.info("Lock acquired: [lock-name=%s]" % lck_name)
```
Comment (wdurairaj) [inline hpedockerplugin/synchronization.py]: I understood from Imran, that this code does'nt print the function name for which the synchronization is required. Requested him to add the function name as part of this logging statement.

File: hpedockerplugin/synchronization.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                return f(*a, **k)
+            except exception.HPEPluginLockFailed:
+                LOG.exception('Failed to acquire lock: [lock-name=%(name)s]',
```
Comment (wdurairaj) [inline hpedockerplugin/synchronization.py]: Same as previous comment. Please add function name as part of logging.

File: hpedockerplugin/synchronization.py (hunk 1 above)
Comment (imran-ansari) [reply hpedockerplugin/synchronization.py]: @wdurairaj  This has been addressed in my second commit. Please have a look.

File: hpedockerplugin/synchronization.py (hunk 2 above)
Comment (imran-ansari) [reply hpedockerplugin/synchronization.py]: @wdurairaj This has been addressed in my second commit. Please have a look.

File: hpedockerplugin/exception.py [hunk 3]
```diff
@@ -212,3 +212,47 @@ class HPEPluginMakeEtcdRootException(PluginException):
...
+class HPEDriverCreateVolumeWithFlashCacheFailed(HPEDriverException):
+    message = ""
+
+
+class HPEDriverVolumeAlreadyExists(HPEDriverException):
+    message = ""
+
+
+class HPEDriverAddVolumeToVvsFailed(HPEDriverException):
+    message = ""
```
Comment (nilangekarss) [inline hpedockerplugin/exception.py]: please check whether it requires a new line at this line?

**Pull request merged**